sdk/eventhub/Azure.Messaging.EventHubs/tests/Core/PartitionContextTests.cs

```diff
@@ -0,0 +1,155 @@
+        /// </summary>
+        ///
+        [Test]
+        public void ConstructorValidatesTheConsumer()
+        {
+            Assert.That(() => new PartitionContext("hub-name", null), Throws.ArgumentNullException);
+        }
+
+        /// <summary>
+        ///   Verifies functionality of the <see cref="PartitionContext.ReadLastEnqueuedEventInformation" />
+        ///   method.
+        /// </summary>
+        ///
+        [Test]
+        public void ReadLastEnqueuedEventInformationDelegatesToTheConsumer()
+        {
+            var lastEvent = new EventData
+            (
+                eventBody: Array.Empty<byte>(),
+                lastPartitionSequenceNumber: 1234,
+                lastPartitionOffset: 42,
+                lastPartitionEnqueuedTime: DateTimeOffset.Parse("2015-10-27T00:00:00Z"),
+                lastPartitionInformationRetrievalTime: DateTimeOffset.Parse("2012-03-04T08:42Z")
+            );
+
+            var eventHubName = "hub-name";
+            var partitionId = "id-value";
+            var mockConsumer = new LastEventConsumerMock(lastEvent);
+            var context = new PartitionContext(eventHubName, partitionId, mockConsumer);
+            var information = context.ReadLastEnqueuedEventInformation();
+
+            Assert.That(information.EventHubName, Is.EqualTo(eventHubName), "The event hub name should match.");
+            Assert.That(information.PartitionId, Is.EqualTo(partitionId), "The partition identifier should match.");
+            Assert.That(information.LastEnqueuedSequenceNumber, Is.EqualTo(lastEvent.LastPartitionSequenceNumber), "The sequence number should match.");
+            Assert.That(information.LastEnqueuedOffset, Is.EqualTo(lastEvent.LastPartitionOffset), "The offset should match.");
+            Assert.That(information.LastEnqueuedTime, Is.EqualTo(lastEvent.LastPartitionEnqueuedTime), "The last enqueue time should match.");
+            Assert.That(information.InformationReceived, Is.EqualTo(lastEvent.LastPartitionInformationRetrievalTime), "The retrieval time should match.");
+        }
+
+        /// <summary>
+        ///   Verifies functionality of the <see cref="PartitionContext.ReadLastEnqueuedEventInformation" />
+        ///   method.
+        /// </summary>
+        ///
+        [Test]
+        public async Task TheConsumerIsNotKeptAlive()
+        {
+            var eventHubName = "hub-name";
+            var partitionId = "id-value";
+            var mockConsumer = new LastEventConsumerMock(new EventData(Array.Empty<byte>()));
+            var context = new PartitionContext(eventHubName, partitionId, mockConsumer);
+
+            // Attempt to clear out the consumer and force GC.
+
+            mockConsumer = null;
+
+            // Because cleanup may be non-deterministic, allow a small set of
+            // retries.
+
+            var attempts = 0;
+            var maxAttempts = 5;
+
+            while (attempts <= maxAttempts)
```

Comment:
This test is a bit sketchy, but so far has proven to be stable in looped runs locally, on Windows and Linux.